Use the new Gtk::Tooltip API instead of the old Gtk::Tooltips API--- a/src/toolbar.cc
+++ b/src/toolbar.cc
@@ -94,26 +94,21 @@
 #endif
 
   // Our tooltips.
-  // NOTE:
-  // http://mail.gnome.org/archives/gtkmm-list/2004-October/msg00010.html
-  // http://mail.gnome.org/archives/gtkmm-list/2004-October/msg00011.html
-  _main.set_tooltips(true);
-  Gtk::Tooltips *tips = _main.get_tooltips_object();
-  _create.set_tooltip(*tips, _("Create a new file"));
-  _open.set_tooltip(*tips, _("Open a file for editing"));
-  _save.set_tooltip(*tips,  _("Save the existing file"));
+  _create.set_tooltip_text(_("Create a new file"));
+  _open.set_tooltip_text(_("Open a file for editing"));
+  _save.set_tooltip_text(_("Save the existing file"));
 #ifdef ENABLE_PRINT
-  _print.set_tooltip(*tips, _("Print this document"));
+  _print.set_tooltip_text(_("Print this document"));
 #endif
-  _close.set_tooltip(*tips, _("Close the active file"));
-  _undo.set_tooltip(*tips, _("Undo"));
-  _redo.set_tooltip(*tips, _("Redo"));
-  _cut.set_tooltip(*tips, _("Cut"));
-  _copy.set_tooltip(*tips, _("Copy"));
-  _paste.set_tooltip(*tips, _("Paste"));
-  _erase.set_tooltip(*tips, _("Delete current selection"));
+  _close.set_tooltip_text(_("Close the active file"));
+  _undo.set_tooltip_text(_("Undo"));
+  _redo.set_tooltip_text(_("Redo"));
+  _cut.set_tooltip_text(_("Cut"));
+  _copy.set_tooltip_text(_("Copy"));
+  _paste.set_tooltip_text(_("Paste"));
+  _erase.set_tooltip_text(_("Delete current selection"));
 #ifdef ENABLE_MAEMO
-  _full_screen.set_tooltip(*tips, _("Toggle full screen mode"));
+  _full_screen.set_tooltip_text(_("Toggle full screen mode"));
 #endif
 
   // This is for both_horiz style
